models/intermediate/int_transform_contract_code.sql

```diff
@@ -0,0 +1,175 @@
+{{
+    config(
+        materialized = 'table',
+        unique_key = ['ledger_key_hash', 'closed_at'],
+        cluster_by = ["ledger_key_hash", "closed_at", "row_hash"],
+        tags = ["soroban_analytics", "intermediate", "daily"]
+    )
+}}
+
+/*
+Model: int_transform_contract_code
+
+Description:
+------------
+This intermediate model combines incremental loading and transformations for contract code data.
+It processes new or updated data, calculates new fields, and performs deduplication.
+
+Load Type: Truncate and Reload
+
+Features:
+---------
+1. Uses execution_date from Airflow to determine the processing window.
+2. Fetches data till the previous day from the Staging layer, one key hash per day (latest).
+3. Calculates `contract_create_ts` and `contract_delete_ts` from staging data.
+4. Joins the calculated timestamps with the incremental data and applies transformations.
+5. Calculates a row hash using SHA256 on all relevant fields.
+6. Performs deduplication based on the row hash of the fully transformed data. This step primarily detects duplicates in consecutive rows.
+
+Usage:
+------
+Compile:
+    dbt compile --models int_transform_contract_code
+
+Run:
+    dbt run --models int_transform_contract_code --full-refresh
+    dbt run --models int_transform_contract_code
+    dbt run --models int_transform_contract_code --vars '{"execution_date": "2024-10-01T00:00:00+00:00"}'
+*/
+
+-- Set the execution date (use Airflow value, fallback to dbt default if absent)
+{% set execution_date = var('execution_date', dbt_airflow_macros.ts()) %}
```

Comment:
You can just use the `dbt_ariflow_macros` instead of making a new var

Reply:
In every dbt run in airflow as it stands right now already sets the var `execution_date`. `dbt_airflow_macros` already uses the `execution_date` variable plus offers date/timestamp casting built in so it's better to use built-ins instead of defining your own variables

Reply:
sure. I will look at the Macro. If it does allow the flexibility to manually pass a date of my choice if and when needed instead of setting it to current timestamp always, will use it.